### ISSUE TITLE
[FR] Nodata value and incomplete ring support

### ISSUE THREAD

**Issue by netthier** [mthh/contour-rs#16]
I want to be able to generate contours that do not include lines on data/nodata borders, 'nodata' being any value either outside the grid or a specifically chosen one inside of it, commonly `f32::MAX` or `f64::MAX`, e.g. (nodata marked red):
Without nodata handling:
![image](https://github.com/mthh/contour-rs/assets/66856670/aa57b7d0-426b-4fd0-b4fe-dde86654f296)

With nodata handling:
![image](https://github.com/mthh/contour-rs/assets/66856670/7048475e-7052-432a-a99d-c2f86f3f0b28)

This also prevents a ring forming around the whole dataset, as seen in https://github.com/mthh/contour-rs/issues/7. While I'm aware of the workaround proposed in the issue, it is not sufficiently performant for me.
I started implementing this feature here: https://github.com/SenseLabsDE/contour-rs/tree/nodata, but the current version is relatively inefficient and results in slowdowns even if the feature is unused.

**Comment by mthh** [comment]
Yes, these are two totally valid use cases that should be supported by this library.

I think it would also be an opportunity to properly support the presence of NaN and (NEG_)INFINITY in the entry grid.
